### ISSUE TITLE
Remove kotlin-reflect dependency

### ISSUE THREAD

**Issue by richarddd** [Litote/kmongo#141]
`KProperty1` could be used without all the other classes from kotlin-reflect.

We could use standard java reflection to deal with registration of pojos etc.

The reason why this is an issue is that it adds lots of complexity.

It also does not work when compiling and image with GraalVMs native-image makes the native-image huge:

```[Thread-3] WARN org.bson.codecs.pojo - Cannot use 'Jedi' with the PojoCodec.
java.lang.IllegalStateException: @NotNull method kotlin/reflect/jvm/internal/impl/builtins/KotlinBuiltIns.getBuiltInClassByFqName must not return null
        at kotlin.reflect.jvm.internal.impl.builtins.KotlinBuiltIns.$$$reportNull$$$0(KotlinBuiltIns.java)
        at kotlin.reflect.jvm.internal.impl.builtins.KotlinBuiltIns.getBuiltInClassByFqName(KotlinBuiltIns.java:357)
```

**Comment by zigzago** [comment]
- kmongo-jackson-mapping depends of third party lib jackson-module-kotlin that uses kotlin-reflect
- kmongo-native-mapping uses a lot kotlin-reflect and it's not so easy to replace it with java reflection 

For my point of view, https://github.com/Litote/kmongo/issues/104 should be implemented first: kotlinx.serialization does not depend of kotlin-reflect

**Comment by richarddd** [comment]
That would be awesome! :)

**Comment by richarddd** [comment]
@zigzago i saw that you implemented https://github.com/Litote/kmongo/issues/104 this weekend 🥳🙇‍♂️🙌 cloud we close this issue as well?

**Comment by zigzago** [comment]
@richarddd Not yet. These are still some reflection involved when checking ids and some other corner cases. (see https://github.com/Litote/kmongo/releases/tag/kmongo-3.11.1 ). Expect a workaround in the next version ;)

**Comment by richarddd** [comment]
@zigzago This should be opened again. All mappers heavily rely on kotlin-reflect. I looked at the source and the main reason why is that all mappers currently need to get the java ower class from a `KProperty1`. We could avoid this completely by using reified operators for all kmongo operations and pass the reified class instance to the mappers. I have created a proof-of-concept branch in my fork where I'm trying this
